float32 x
float32 y
float64 theta
float64 vel
bool correct
